public/app/core/components/TimeSeries/utils.ts

```diff
@@ -178,15 +178,18 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn = ({
   } else {
     let scaleDistr: ScaleDistributionConfig = { ...xField.config.custom?.scaleDistribution };
 
-    if ((xField.config.min ?? xField.values[0]) === 0 && scaleDistr?.type === 'log') {
-      scaleDistr.type = ScaleDistribution.Linear;
-    }
-
     builder.addScale({
       scaleKey: xScaleKey,
       orientation: isHorizontal ? ScaleOrientation.Horizontal : ScaleOrientation.Vertical,
       direction: isHorizontal ? ScaleDirection.Right : ScaleDirection.Up,
-      range: (u, dataMin, dataMax) => [xField.config.min ?? dataMin, xField.config.max ?? dataMax],
+      range: (u, dataMin, dataMax) => {
+        const min = xField.config.min ?? dataMin;
+        if (min <= 0 && scaleDistr?.type === 'log') {
+          return [null, null];
```

Comment:
you'll probably want `uPlot.rangeLog()`

https://github.com/grafana/grafana/blob/1d2f271c9550a07fac68b7f176c06e48ed2f0b7d/packages/grafana-ui/src/components/uPlot/config/UPlotScaleBuilder.ts#L201